src/celestia/celx_internal.h

```diff
@@ -149,8 +308,41 @@ class CelxLua
                         const char* errorMessage = "Boolean argument expected",
                         bool defaultValue = false);
 
-    LuaState* getLuaStateObject();
+    template<typename T> T *safeGetUserData(int index = 0,
+                                            FatalErrors errors = AllErrors,
+                                            const char *errorMessage = "User data expected")
+    {
+        if (!safeIsValid(index))  return nullptr;
```

Comment:
Please don't use such style. There are of places in your code where it can be misunderstood due to this style.

Reply:
Do you mean `if - then` in one line?

Reply:
Yes